modules/p2/ding_list/include/form.inc

```diff
@@ -741,3 +741,52 @@ function ding_list_list_permissions_form_share_callback($form, &$form_state) {
 
   return $response;
 }
+
+/**
+ * Ask user to reset their lists.
+ */
+function ding_list_reset_user_form($form, &$form_state, $account) {
+  $form['warning'] = array(
+    '#markup' => '<p>' . t('Are you sure you want to reset all your lists.') . '</p>',
+  );
+
+  $form['actions'] = array(
+    '#type' => 'actions',
+    'yes' => array(
+      '#type' => 'submit',
+      '#value' => t('Yes'),
+    ),
+    'cancel' => array(
+      '#type' => 'submit',
+      '#value' => t('Cancel'),
+      '#submit' => array('ding_list_reset_user_form_submit_cancel'),
+      '#limit_validation_errors' => array(),
+    ),
+  );
+
+  return $form;
+}
+
+/**
+ * Cancel handler for ding_list_reset_user_form.
+ */
+function ding_list_reset_user_form_submit_cancel($form, &$form_state) {
+  $account = $form_state['build_info']['args'][0];
+  $form_state['redirect'] = 'user/' . $account->uid . '/my-library';
+}
+
+/**
+ * Submit handler for ding_list_reset_user_form.
+ */
+function ding_list_reset_user_form_submit($form, &$form_state) {
+  $account = $form_state['build_info']['args'][0];
+
+  ding_list_reset_user($account);
+
+  ding_list_setup_autocreated($account);
+  if (ding_user_consent_has_consent('loan_history_store')) {
+    ding_list_update_historical_loans($account);
+  }
```

Comment:
Is there a reason why these four lines are not part of `ding_list_reset_user()`?

If we at any other point need to reset a user it would be nice to have a single function to achieve that. One might think that `ding_list_reset_user()` would be sufficient but that is not the case.